### PR TITLE
Makes bottles not as terrible

### PR DESCRIPTION
**Pull request by IcyV**
Previously, the brown rbottles found in chemical lockers were stuck as
brown forever. Now they actually display the chemical inside of them.

The bottles for items such as morphine or toxins that you could get from
vendors always had a specific bottle that never updated based on
chemical or volume. These have new bottles and properly work now.

bottle images

Much thanks to FullOfSkittles for the bottle sprite and KasparoVv for
pretty much most of getting this worked out since I'm terrible with sprites.

:cl: IcyV
add: Adds new bottle sprite
fix: Fixes many bottles being locked to a single icon
/:cl:

### CODE REVIEW

**Comment by Regen1** [comment]
The brown bottles should probably have a brown tint to them, right now it just looks like the green is added on top of it

**Comment by FlattestGuitar** [comment]
Adding more transparency to the overlays would help, yeah.

**Comment by IcyV** [comment]
I'll take a poke at that and try to get it done here when I'm more awake.

**Comment by IcyV** [comment]
Updated the main post with a new picture on top of sprucing up how this works.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 1]
```diff
@@ -15,16 +16,18 @@
 /obj/item/weapon/reagent_containers/glass/bottle/New()
 	..()
 	if(!icon_state)
-		icon_state = "bottle[rand(1,20)]"
+		icon_state = pick("reagent_bottle","rounded_bottle")
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: reagent bottles are distinct from regular bottles, especially since they contain nearly double the volume--probably shouldn't be setting this as a potential option.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 2]
```diff
@@ -40,7 +43,7 @@
 				filling.icon_state = "[icon_state]100"
 
 		filling.icon += mix_color_from_reagents(reagents.reagent_list)
-		overlays += filling
+		underlays += filling
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: Why was this changed...to an underlay?

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 3]
```diff
@@ -728,6 +727,7 @@
 			buffer_reagents_list[++buffer_reagents_list.len] = list("name" = R.name, "volume" = R.volume, "id" = R.id, "description" = R.description)
 
 	data["pillsprite"] = pillsprite
+	data["bottlesprite"] = bottlesprite
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Machinery.dm]: This doesn't appear to be plugged in to ...well, anywhere.

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 4]
```diff
@@ -667,14 +667,13 @@
 			usr << browse(dat, "window=chem_master_iconsel;size=225x193")
 			return
 		else if(href_list["change_bottle"])
-			#define MAX_BOTTLE_SPRITE 20 //max icon state of the bottle sprites
 			var/dat = "<table>"
 			var/j = 0
-			for(var/i = 1 to MAX_BOTTLE_SPRITE)
+			for(var/i in list("rounded_bottle"))
```
Comment (Fox-McCloud) [inline code/modules/reagents/Chemistry-Machinery.dm]: why is this a list when you only have a single sprite now?

File: code/modules/client/asset_cache.dm [hunk 5]
```diff
@@ -268,8 +268,8 @@ proc/getFilesSlow(var/client/client, var/list/files, var/register_asset = TRUE)
 /datum/asset/chem_master/register()
 	for(var/i = 1 to 20)
 		assets["pill[i].png"] = icon('icons/obj/chemical.dmi', "pill[i]")
-	for(var/i = 1 to 20)
-		assets["bottle[i].png"] = icon('icons/obj/chemical.dmi', "bottle[i]")
+	for(var/i in list("rounded_bottle"))
```
Comment (Fox-McCloud) [inline code/modules/client/asset_cache.dm]: why is this a list when you only have a single sprite now?

**Comment by Fox-McCloud** [comment]
There...only appears to be one bottle sprite now? the rounded ones with an overlay of liquid?

**Comment by IcyV** [comment]
So a good handful of those were because a separate thing I was working on in another branch wormed its way in somehow on top of it when I started and I didn't notice. I'll fix that all up.

**Comment by IcyV** [comment]
Also apparently KasparoVv is a hero and did it for me in a better way.

File: code/modules/reagents/Chemistry-Machinery.dm (hunk 3 above)
Comment (KasparoVy) [inline code/modules/reagents/Chemistry-Machinery.dm]: Now plugs into the nanoui template

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 2 above)
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/bottle.dm]: To address the complaints that coloured bottles (mainly the reagent one) didn't tint the reagents inside it, also because most of the bottle sprites have these little 'label' pixels that would normally just get hidden

**Comment by Fox-McCloud** [comment]
Sorry for not getting this merged in as soon it *should* have been. Been busy, of late.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 6]
```diff
@@ -15,16 +16,18 @@
 /obj/item/weapon/reagent_containers/glass/bottle/New()
 	..()
 	if(!icon_state)
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: This won't even be possible anymore given that the `icon_state` for the base type is explicitly set.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 7]
```diff
@@ -98,107 +95,92 @@
...
 	name = "Reagent Bottle"
 	desc = "A bottle for storing reagents"
-	icon_state = "rbottle"
+	icon_state = "reagent_bottle"
 	volume = 50
 
 /obj/item/weapon/reagent_containers/glass/bottle/reagent/oil
 	name = "Oil Bottle"
 	desc = "A reagent bottle. Contains oil."
-	icon_state = "rbottle1"
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: All of these used specific icons to ensure that they all stack up nice and neat in the cabinet (not a very good way of doing it); you'll want to adjust their `pixel_x` and `pixel_y` to retain that behavior.

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 8]
```diff
@@ -15,16 +16,18 @@
...
 /obj/item/weapon/reagent_containers/glass/bottle/on_reagent_change()
 	update_icon()
 
 /obj/item/weapon/reagent_containers/glass/bottle/update_icon()
 	overlays.Cut()
+	underlays.Cut()
 
 	if(reagents.total_volume)
-		var/image/filling = image('icons/obj/reagentfillings.dmi', src, "[icon_state]10")
+		var/image/filling = image('icons/obj/reagentfillings.dmi', src, "bottle10")
```
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: this means every single bottle type will use the same filling.

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 2 above)
Comment (Fox-McCloud) [inline code/modules/reagents/reagent_containers/bottle.dm]: It would honestly be better to shade the filling differently to have that effect rather than to uh, well...utilize underlays.

**Comment by Fox-McCloud** [comment]
There's a few issues left---it's pretty close though; hopefully with @KasparoVy's help, this can get merged in soon.

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 2 above)
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/bottle.dm]: Then we'd have the bottle 'label' pixels getting the same colour as the reagents, making them useless again. 
bottlelabels

The underlays elegantly accomplish the desired graphical effect with the simplest sprite, and it makes sense to have the chemical filler 'beneath' the bottle icon as the filler is supposed to represent the volume of liquid inside (behind the face of) the bottle.
I just feel like it's not much worth the effort to tint the sprites up when the underlays (and the one additional line of code that comes with it) do the job

File: code/modules/reagents/reagent_containers/bottle.dm [hunk 9]
```diff
@@ -1,11 +1,12 @@
 
 //Not to be confused with /obj/item/weapon/reagent_containers/food/drinks/bottle
 
+
 /obj/item/weapon/reagent_containers/glass/bottle
 	name = "bottle"
 	desc = "A small bottle."
 	icon = 'icons/obj/chemical.dmi'
-	icon_state = null
+	icon_state = "rounded_bottle"
```
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/bottle.dm]: Woops, just supposed to be 'round_bottle'.

File: code/modules/reagents/reagent_containers/bottle.dm (hunk 7 above)
Comment (KasparoVy) [inline code/modules/reagents/reagent_containers/bottle.dm]: Should be resolved now, thanks for catching that
bottlesstackingproperly
